react/components/SearchResult.js

```diff
@@ -34,9 +37,12 @@ export default class SearchResult extends Component {
 
   handleLayoutChange = (e, mode) => {
     e.preventDefault()
+    const defaultModes = ['small', 'inline', 'normal']
+    const modeIndex = defaultModes.indexOf(this.state.galleryLayoutMode) !== -1 ? (defaultModes.indexOf(this.state.galleryLayoutMode) + 1) % 3 : 0
```

Comment:
I suggest destructuring the gallery LayoutMode to avoid repetition. Will defaultModes.indexOf(this.state.gallery LayoutMode) anytime be -1?